grant/redeem.go

```diff
@@ -83,26 +88,16 @@ func (service *Service) Consume(ctx context.Context, walletInfo wallet.Info, tra
 		// 1. Enforce transaction checks and verify transaction signature
 		// NOTE for uphold provider we currently check against user provided publicKey
 		//      thus this check does not protect us from a valid fake signature
-		txInfo, err := userWallet.VerifyTransaction(transaction)
+		txInfo, err := userWallet.VerifyAnonCardTransaction(transaction)
 		if err != nil {
 			return nil, err
 		}
-		if *txInfo.AltCurrency != altcurrency.BAT {
-			return nil, errors.New("only grants submitted with BAT transactions are supported")
-		}
```

Comment:
perhaps this validation is elsewhere, but do we not want to check for negative values? if only to give users more info if something goes wrong

Reply:
This seems reasonable, Going to defer this to @evq

Reply:
It was moved to `VerifyAnonCardTransaction` https://github.com/brave-intl/bat-go/pull/226/files/31621281d300607c9e97f8f8fecb0ab2975196da#diff-b1a039efc16d53e8905f02be7d450613R529 ?